core/shared/platform/common/posix/posix_thread.c

```diff
@@ -543,14 +543,11 @@ signal_callback(int sig_num, siginfo_t *sig_info, void *sig_ucontext)
     if (prev_sig_act && (prev_sig_act->sa_flags & SA_SIGINFO)) {
         prev_sig_act->sa_sigaction(sig_num, sig_info, sig_ucontext);
     }
-    else if (prev_sig_act && (void *)prev_sig_act->sa_handler == SIG_DFL) {
-        /* Default action */
-        sigaction(sig_num, prev_sig_act, NULL);
-    }
-    else if (prev_sig_act && (void *)prev_sig_act->sa_handler == SIG_IGN) {
-        /* Ignore this signal */
-    }
-    else if (prev_sig_act && prev_sig_act->sa_handler) {
+    else if (prev_sig_act
+             && prev_sig_act->sa_handler
+             /* Ignore SIG_DFL and SIG_IGN */
+             && (void *)prev_sig_act->sa_handler != SIG_DFL
+             && (void *)prev_sig_act->sa_handler != SIG_IGN) {
```

Comment:
do you mean to abort() if it's SIG_DFL/IGN?

Reply:
Yes, do you have better idea?

Reply:
no. it's fine.
i asked because it was not obvious from the description "Ignore handling SIG_DFL/SIG_IGN for previous sig action
".

Reply:
Yes, it makes a little confusing, I updated the comment.